### ISSUE TITLE
Model compression

### ISSUE THREAD

**Issue by hammer** [hammerlab/mhcflurry#5]
Can we use [HashedNets](http://arxiv.org/abs/1504.04788) or [deep compression](http://arxiv.org/abs/1510.00149) to make it easier to distribute the parameters for our model to end users?

**Comment by hammer** [comment]
Would help with #2

**Comment by iskandr** [comment]
I don't think that model compression, at least naively implemented on top of a Theano or TensorFlow based library, will actually help much with model size. They both will probably rely on having dense weight matrices (and using masks during training to keep track of which weights are disabled). We could probably write custom fprop code for sparse weight arrays.

**Comment by hammer** [comment]
New model compression technique: [Dynamic Network Surgery for Efficient DNNs](http://arxiv.org/abs/1608.04493)

**Comment by timodonnell** [comment]
Two recent papers from the Bengio group on using low-precision weight matrices (even binary):

 * [Quantized Neural Networks](https://arxiv.org/pdf/1609.07061.pdf)
 * [Recurrent Neural Networks With Limited Numerical Precision]( https://arxiv.org/pdf/1608.06902.pdf)

**Comment by timodonnell** [comment]
In the latest release our serialized models just got much smaller in terms of file size (full download is ~40mb) so this is less of a concern, closing for now

**Comment by hammer** [comment]
For future reference, TensorFlow now has a model optimization toolkit 
https://medium.com/tensorflow/introducing-the-model-optimization-toolkit-for-tensorflow-254aca1ba0a3?linkId=57036398
